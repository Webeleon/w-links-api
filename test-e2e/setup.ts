import { Test, TestingModule } from '@nestjs/testing';
import { AppModule } from '../src/app.module';
import { databaseConfig } from '../src/configurations/database.config';
import { INestApplication } from '@nestjs/common';
import { Repository } from 'typeorm';
import { UsersEntity } from '../src/users/users.entity';
import { getRepositoryToken } from '@nestjs/typeorm';

export const getTestingApp = async (): Promise<INestApplication> => {
  const moduleFixture: TestingModule = await Test.createTestingModule({
    imports: [AppModule],
  })
    .overrideProvider(databaseConfig.KEY)
    .useValue({
      type: 'better-sqlite3',
      database: ':memory:',
    })
    .compile();

  const app = moduleFixture.createNestApplication();
  await app.init();

  return app;
};

