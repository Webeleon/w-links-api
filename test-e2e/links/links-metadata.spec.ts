import { INestApplication } from '@nestjs/common';
import * as request from 'supertest';
import { getTestingApp } from '../setup';

describe('Links metadata (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    app = await getTestingApp();
  });

  afterEach(async () => {
    await app.close();
  });

  it('/links/metadata/types (GET)', async () => {
    const { status, body } = await request(app.getHttpServer()).get(
      '/links/metadata/link-types',
    );
    expect(status).toBe(200);
    expect(body).toMatchSnapshot();
  });
});
