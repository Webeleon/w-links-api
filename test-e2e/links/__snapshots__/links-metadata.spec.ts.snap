// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links metadata (e2e) /links/metadata/types (GET) 1`] = `
Array [
  "discord",
  "youtube",
  "tiktok",
  "linkedin",
  "devto",
  "github",
  "twitch",
  "other",
]
`;
